frontend/src/views/RoleView.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <div class="PageWrapper">
+    <div class="TableContainer">
+      <h1>{{ roleName }}</h1>
+      <h2>Attached Knowledges</h2>
+      <br />
+      <ElTable :data="knowledges" style="width: 100%">
+        <ElTableColumn prop="name" label="Knowledges" />
+        <ElTableColumn fixed="right" label="Actions" width="150">
+          <template #default="{ row }">
+            <div>
+              <ElButton @click="onDeleteRow(row)" type="text" size="small">Delete</ElButton>
+              <ElButton @click="onEditRow(row)" type="text" size="small">Edit</ElButton>
+            </div>
+          </template>
+        </ElTableColumn>
+      </ElTable>
+      <div class="demo">
+        <br />
+        <ElButton type="primary" @click="showDialogAddNewKnowledge">Add new Knowledge</ElButton>
+        <ElDialog v-model="createKnowledgeDialogFormVisible" title="New Knowledge" width="500">
+          <ElInput v-model="newKnowledgeName" placeholder="Knowledge name" :clearable="false" />
+          <template #footer>
+            <div class="dialog-footer">
+              <ElButton @click="createKnowledgeDialogFormVisible = false">Cancel</ElButton>
+              <ElButton type="primary" @click="createKnowledge">Confirm</ElButton>
+            </div>
+          </template>
+        </ElDialog>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted } from 'vue'
+import { ElTable, ElTableColumn, ElButton, ElDialog, ElInput } from 'element-plus'
+import { ErrorMessage, SuccessMessage } from '@/services/messages/messages'
+import type { Knowledge } from '@/domain/Knowledge'
+import { useKnowledgeStore } from '@/stores/knowledges/index'
+import { useRolesStore } from '@/stores/roles/index'
+
+const onDeleteRow = (row: Knowledge) => {
+  console.log('Detail clicked for:', row)
+}
+
+const onEditRow = (row: Knowledge) => {
+  console.log('Edit clicked for:', row)
+}
+
+const knowledgesStore = useKnowledgeStore()
+const roleStore = useRolesStore()
+const createKnowledgeDialogFormVisible = ref(false)
+const newKnowledgeName = ref('')
+const roleId = ref<string>('1')
+const roleName = ref<string>('')
+const knowledges = ref<Knowledge[]>([])
+
+onMounted(async () => {
+  const role = await roleStore.fetchRoleById(roleId.value)
+  knowledges.value = await knowledgesStore.fetchKnowledgesByRoleId(parseInt(roleId.value))
```

Comment:
If you pass a number to your ref, you don't need this cast

```suggestion
  knowledges.value = await knowledgesStore.fetchKnowledgesByRoleId(roleId.value)
```